store/store_test.go

```diff
@@ -6898,30 +6898,52 @@ func (s *storeTestSuite) TestSnapActionRefreshesBothAuths(c *C) {
 }
 
 func (s *storeTestSuite) TestConnectivityCheck(c *C) {
+	expectedConn := make(map[string]bool, 4)
+	var mockServerURL *url.URL
 	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		io.WriteString(w, "all good")
+		switch r.URL.Path {
+		case "/acl/":
+			c.Check(r.Method, Equals, "GET")
+			w.WriteHeader(401)
+			io.WriteString(w, `{"error_list": [{"code": "macaroon-permission-required"}]}`)
+		case "/u1api":
+			c.Check(r.Method, Equals, "HEAD")
+			w.WriteHeader(200)
+		case "/api/v1/snaps/details/core":
+			c.Check(r.Method, Equals, "GET")
+			c.Check(r.URL.Query().Encode(), DeepEquals, "fields=anon_download_url")
+			u, err := url.Parse("/download/core")
+			c.Assert(err, IsNil)
+			io.WriteString(w, fmt.Sprintf(`{"anon_download_url": %q}`, mockServerURL.ResolveReference(u).String()))
+		case "/download/core":
+			c.Check(r.Method, Equals, "HEAD")
+			w.WriteHeader(200)
+		default:
+			c.Fatalf("unexpected request: %s", r.URL.String())
+			return
+		}
+		expectedConn[mockServerURL.ResolveReference(r.URL).String()] = true
```

Comment:
This feels a little odd, shouldn't we define outside what we expect and not build the expected list as part of what the code triggers? I mean, right now if we don't do a HEAD requqest on /download/core we would not know given the current test? Or am I missing something?

Reply:
you're probably right. fixed.